test/basic/test_utils.jl

```diff
@@ -8,3 +13,49 @@ using LightGBM
     expected = "categorical_feature=0,2,4,6,8"
     @test occursin(expected, ds_parameters)
 end
+
+
+
+@testset "loadmodel predicts same as original model -- regression" begin
```

Comment:
Nice tests. To make this more rigorous, I would  load the model onto a newly constructed default estimator and show predictions come out the same. Maybe use more non-default params for the original estimator too (so we can in fact spot/verify that predictions are invariant to the other params)